libs/common/src/admin-console/services/organization/organization.service.spec.ts

```diff
@@ -1,196 +1,328 @@
-import { UserId } from "../../../types/guid";
+import { OrganizationId, UserId } from "../../../types/guid";
 import { OrganizationData } from "../../models/data/organization.data";
+import { Organization } from "../../models/domain/organization";
 
 import { OrganizationService, ORGANIZATIONS } from "./organization.service";
 
-describe("Organization Service", () => {
+describe("OrganizationService", () => {
   let organizationService: OrganizationService;
 
-  let stateService: MockProxy<StateService>;
-  let activeAccount: BehaviorSubject<string>;
-  let activeAccountUnlocked: BehaviorSubject<boolean>;
-
-  const mockUserId = Utils.newGuid() as UserId;
-  let accountService: FakeAccountService;
-  let stateProvider: FakeStateProvider;
-  let activeUserOrganizationsState: FakeActiveUserState<Record<string, OrganizationData>>;
-
-  const resetStateService = async (
-    customizeStateService: (stateService: MockProxy<StateService>) => void,
-  ) => {
-    mockClear(stateService);
-    stateService = mock<StateService>();
-    stateService.activeAccount$ = activeAccount;
-    stateService.activeAccountUnlocked$ = activeAccountUnlocked;
-    customizeStateService(stateService);
-    organizationService = new OrganizationService(stateService, stateProvider);
-    await new Promise((r) => setTimeout(r, 50));
-  };
-
-  function prepareStateProvider(): void {
-    accountService = mockAccountServiceWith(mockUserId);
-    stateProvider = new FakeStateProvider(accountService);
+  const fakeUserId = Utils.newGuid() as UserId;
+  let fakeAccountService: FakeAccountService;
+  let fakeStateProvider: FakeStateProvider;
+  let fakeActiveUserState: FakeActiveUserState<Record<string, OrganizationData>>;
+
+  /**
+   * It is easier to read arrays than records in code, but we store a record
+   * in state. This helper methods lets us build organization arrays in tests
+   * and easily map them to records before storing them in state.
+   */
+  function arrayToRecord(input: OrganizationData[]): Record<OrganizationId, OrganizationData> {
+    if (input == null) {
+      return undefined;
+    }
+    return Object.fromEntries(input?.map((i) => [i.id, i]));
   }
 
-  function seedTestData(): void {
-    activeUserOrganizationsState = stateProvider.activeUser.getFake(ORGANIZATIONS);
-    activeUserOrganizationsState.nextState({ "1": organizationData("1", "Test Org") });
+  /**
+   * There are a few assertions in this spec that check for array equality
+   * but want to ignore a specific index that _should_ be different. This
+   * function takes two arrays, and an index. It checks for equality of the
+   * arrays, but splices out the specified index from both arrays first.
+   */
+  function isEqualExceptForIndex(x: any[], y: any[], indexToExclude: number) {
+    // Clone the arrays to avoid modifying the reference values
+    const a = [...x];
+    const b = [...y];
+    delete a[indexToExclude];
+    delete b[indexToExclude];
+    return a === b;
   }
```

Comment:
I clearly blanked on this when we were discussing it. Because you're using `===` instead of Jest's `toEqual`, it's checking reference equality instead of value equality. This will always return false, which I'm assuming it is because your tests are failing.

I suggest moving the assertion inside this function:

```suggestion
  function isEqualExceptForIndex(x: any[], y: any[], indexToExclude: number) {
    // Clone the arrays to avoid modifying the reference values
    const a = [...x];
    const b = [...y];
    delete a[indexToExclude];
    delete b[indexToExclude];
    expect(a).toEqual(b);
  }
```

and then calling it on its own (not part of a jest expect statement):

```ts
// other assertions
isEqualExceptForIndex(a, b, 5);
```

Renaming it to `expectIsEqualExceptForIndex` might make that usage a bit clearer.

Reply:
Good catch, implemented as suggested.